src/components/RepositoryDetail/RepositoryDetail.tsx

```diff
@@ -0,0 +1,366 @@
+              <FontAwesomeIcon icon={faSignInAlt} />
+              &nbsp;
+              Go to Repository
+            </Button>
+        )}
+      { repo.works && (repo.works.totalCount>0) && (
+          <Link href={"/doi.org?query=client.uid:" + repo.id}>
+          <Button block bsStyle="primary">
+              <FontAwesomeIcon icon={faNewspaper} />
+              &nbsp;
+              Find Related Works
+            </Button>
+          </Link>
+        )}
+      </>
+    )
+  }
+
+  const contacts = () => {
+    return (
+      <>
+      { repo.contacts && (
+        <h3>Contacts</h3>
+      )}
+    </>
+    )
+  }
+
+  const shareDisplay = () => {
+    const info = pageInfo(repo);
+    return (
+      <>
+        <h3>Share</h3>
+        <EmailShareButton url={info.pageUrl} title={info.title}>
+          <FontAwesomeIcon icon={faEnvelope} /> Email
+        </EmailShareButton>
+        <TwitterShareButton url={info.pageUrl} title={info.title}>
+          <FontAwesomeIcon icon={faTwitter} /> Twitter
+        </TwitterShareButton>
+        <FacebookShareButton url={info.pageUrl} title={info.title}>
+          <FontAwesomeIcon icon={faFacebook} /> Facebook
+        </FacebookShareButton>
+      </>
+    )
+  }
+  return (
+    <>
+        <div className={styles.gotoButtons}>
+          {gotoButtons()}
+        </div>
+        <div className={styles.contacts}>
+          {contacts()}
+        </div>
+        <div className={styles.share}>
+          {shareDisplay()}
+        </div>
+      </>
+    )
+
+}
+export const RepositoryDetail: React.FunctionComponent<Props> = ({
+  repo
+}) => {
+
+
+
+  const dashboard = () => {
+    return (
+      <>
+      <h3>{compactNumbers(repo.works.totalCount)} Deposits</h3>
+
+      <div className={styles.grid}>
+        <ConditionalProductionChart 
+          title="Year of Publication"
+          data={facetToData(repo.works.published)}
+        />
+        <ConditionalDonutChart
+          data={facetToData(repo.works.resourceTypes)}
+          count={repo.works.totalCount}
+          title="Deposit Type"
+        />
+        <ConditionalBarChart title="Top Depositors" data={repo.works.authors} />
+        <ConditionalBarChart title="Fields of Science" data={repo.works.fieldsOfScience} />
+        <ConditionalBarChart title="Deposit Languages" data={repo.works.languages} />
+        <ConditionalBarChart title="Deposit Licenses" data={repo.works.licenses} />
+      </div>
+      </>
+    )
+  }
+
+  const tags = () => {
+    return "TAGS"
+  }
+
+  const advise = () => {
+    return (
+      <>
+        <p>
+          If you plan to deposit your research data in this repository, 
+          { repo.url && (
+            <>
+              &nbsp; go to <a href={repo.url}>{repo.url}</a>
+            </>
+          )}
+          { repo.url && (
+            <>
+              &nbsp; or contact the repository for more information via <a href={repo.url}>{repo.url}</a>
+            </>
```

Comment:
Is this a conditional to check for the existence / non-nullity of `repo.url` before rendering the contained bits of template?

If so, two things spring to mind:

1) you could probably optimise it slightly into a single check,

2) if `repo.url` is null, will this just render "If you plan to deposit your research data in this repository," and then the "More Information" section? If yes, then it's probably worth putting that phrase inside the conditional as well

Reply:
Combined.  The second repo.url conditional would eventually be replaced by contact information when we get that to return from the gql endpoint.  But for now I will just remove and simplify.